src/App/Controller/PrivateDiscussionController.php

```diff
@@ -0,0 +1,325 @@
+                $em->persist($post);
+                $em->flush();
+
+                return $this->redirect($this->generateUrl('private_discussion_show', [
+                    'slug' => $discussion->getSlug(),
+                ]));
+            }
+
+            return $this->render('App:PrivateDiscussion:show.html.twig', [
+                'discussion' => $discussion,
+                'form'       => $form->createView(),
+                ]);
+        }
+
+        return $this->render('App:PrivateDiscussion:show.html.twig', [
+            'discussion' => $discussion,
+        ]);
+    }
+
+    /**
+     * @Security("has_role('ROLE_USER')")
+     */
+    public function addAction(Request $request)
+    {
+        $discussion = new PrivateDiscussion();
+        $user = $this->getUser();
+        $form = $this->createForm(new AddPrivateDiscussionType($user->getId()), $discussion);
+        $form->handleRequest($request);
+
+        if ($form->isValid()) {
+            $discussion->setAdmin($user)->resetUnreaders();
+            $em = $this->getDoctrine()->getManager();
+            $em->persist($discussion);
+            $em->flush();
+
+            return $this->redirect($this->generateUrl('private_discussion_show', [
+                'slug' => $discussion->getSlug(),
+            ]));
+        }
+
+        return $this->render('App:PrivateDiscussion:form.html.twig', [
+            'form' => $form->createView(),
+        ]);
+    }
+
+    /**
+     * @Security("has_role('ROLE_USER')")
+     */
+    public function editAction(Request $request, $slug)
+    {
+        $discussion = $this->getDoctrine()->getRepository('App:PrivateDiscussion')->findOneBySlug($slug);
+
+        if (null ===  $discussion) {
+            throw $this->createNotFoundException();
+        }
+
+        if (false === $this->get('security.context')->isGranted('edit', $discussion)) {
+            throw new AccessDeniedException();
+        }
+
+        $form = $this->createForm(new EditDiscussionType(), $discussion);
+        $form->handleRequest($request);
+
+        if ($form->isValid()) {
+            $em = $this->getDoctrine()->getManager();
+            $em->persist($discussion);
+            $em->flush();
+
+            return $this->redirect($this->generateUrl('private_discussion_show', [
+                'slug' => $discussion->getSlug(),
+             ]));
+        }
+
+        return $this->render('App:PrivateDiscussion:form.html.twig', [
+            'form' => $form->createView(),
+        ]);
+    }
+
+    /**
+     * @Security("has_role('ROLE_USER')")
+     */
+    public function addMemberAction(Request $request, $slug)
+    {
+        $discussion = $this->getDoctrine()->getRepository('App:PrivateDiscussion')->findOneBySlug($slug);
+
+        if (null ===  $discussion) {
+            throw $this->createNotFoundException();
+        }
+
+        $users         = $this->getDoctrine()->getRepository('App:User')->findAll();
```

Comment:
Waoutch. Here you just retrieve *all* the users from the database, just to… know how many they are.
You can create a method in your repository to just return an int with the number of users.